packages/core/event.js

```diff
@@ -181,6 +193,19 @@ Event.create = function (maybeError, tolerateNonErrors, handledState, component,
   if (error.name === 'InvalidError') {
     event.addMetadata(`${component}`, 'non-error parameter', makeSerialisable(maybeError))
   }
+  if (error.cause) {
+    const causes = getCauseStack(maybeError).slice(1)
+    const normalisedCauses = map(causes, (cause) => {
+      const [error] = normaliseError(cause, tolerateNonErrors, 'error cause')
+      if (error.name === 'InvalidError') {
+        event.addMetadata('error cause', makeSerialisable(cause))
+      }
+      return error
+    })
+    const causeErrors = map(normalisedCauses, (cause) => createBugsnagError(cause.name, cause.message, Event.__type, [])) // do not include stacktrace for error causes
```

Comment:
If the cause is a JS `Error` it's fine to include the stacktrace.  It's only if it is some other type we don't want to generate it.  But looking at the implementation that generation is only done in `Event.create()` at the top level anyway.  The stack should already be on the normalised error I think?  It's hard to follow it completely.

As related issue, looking at the `normaliseError` function, do we need to do something different with the `internalFrames` if it's a cause?  I don't know the answer but worth checking.

Reply:
refactored the normalisation of the causes, which checks for the existence of an error stack before passing to `normaliseError` and have updated test cases to ensure stacks are present where expected.